src/client/App.tsx

```diff
@@ -186,33 +169,43 @@ function App() {
     setTheme,
     setIsSettingsOpen,
     closeSettings,
-    editorNoLongerWantsFocus,
-    setUsername,
   } = useActions(dispatch);
 
   // The set of open directories.
   // TODO move this into reducer/useActions
   const { isDirectoryOpen, toggleDirectory } =
     useOpenDirectories();
 
-  usePersistUsername(username);
-
   // Cache of CodeMirror editors by file id.
   const editorCache: EditorCache = useEditorCache();
 
+  // A function that will focus the editor.
+  const focusEditor = useCallback(
+    (fileId: FileId) => {
+      // TODO figure out a way to do this without the timeout
+      setTimeout(() => {
```

Comment:
It looks like you started from an older version of the code, and this PR is bringing back these old bits.